klib/Makefile

```diff
@@ -0,0 +1,20 @@
+PROGRAMS=	test
+
+SRCS-test= \
+	$(CURDIR)/test.c
+
+all: $(PROGRAMS)
+
+include ../rules.mk
+
+ifeq ($(UNAME_s),Darwin)
+ELF_TARGET=     -target x86_64-elf
+CFLAGS+=        $(ELF_TARGET)
+LD=             x86_64-elf-ld
+OBJDUMP=        x86_64-elf-objdump
+else
+LD=             $(CROSS_COMPILE)ld
+endif
+
+CFLAGS+=	$(KERNCFLAGS) -I$(CURDIR) -fPIC -DKLIB
```

Comment:
We may also need to remove `-ffunction-sections` from KERNCFLAGS and add it here, otherwise kernel functions that are never called inside the kernel will be discarded by the linker and so won't be found by kernel modules. For example, in the TLS module I need to call `direct_connect()`, which, being a new function (introduced in commit https://github.com/nanovms/nanos/commit/f8b445194f697c79d3319c4f94c7103096819f90), is never used in other parts of the kernel.
There are also other functions (such as strstr, calloc, free, rand, snprintf) which I'm now implementing inside the TLS module by could potentially be in the main kernel.